resources/assets/components/pages/AccountPage/VoterReg/VoterRegStatusBlock.js

```diff
@@ -1,47 +1,107 @@
 import React from 'react';
+import { get } from 'lodash';
 import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { useQuery } from '@apollo/react-hooks';
 
 import checkmark from './checkmark.svg';
-import rejectedIcon from './rejectedIcon.svg';
-import pendingIcon from './pendingIcon.svg';
+import {
+  getTrackingSource,
+  needToVerifyVoterRegStatuses,
+  getCheckRegistrationStatusURL,
+  verifiedCompletedVoterRegStatuses,
+  verifiedIneligibleVoterRegStatuses,
+  USER_VOTER_REGISTRATION_STATUS_QUERY,
+} from '../../../../helpers/voter-registration';
+import Spinner from '../../../artifacts/Spinner/Spinner';
+import ErrorBlock from '../../../blocks/ErrorBlock/ErrorBlock';
 
 import './voter-reg.scss';
 
-const VoterRegStatusBlock = ({ status }) => {
-  if (status === 'PENDING') {
-    return (
-      <div className="voter-reg -yellow flex items-center">
-        <img className="post-badge icon-clock" src={pendingIcon} alt="hello" />
-        <div className="ml-3">Your voter reg status is pending!</div>
-      </div>
-    );
+const VoterRegStatusBlock = ({ userId }) => {
+  const { loading, error, data } = useQuery(
+    USER_VOTER_REGISTRATION_STATUS_QUERY,
+    { variables: { userId } },
+  );
+
+  const registrationStatus = get(data, 'user.voterRegistrationStatus', null);
+
+  const verifiedRegistrationStatus =
+    verifiedIneligibleVoterRegStatuses(registrationStatus) ||
+    verifiedCompletedVoterRegStatuses(registrationStatus);
+
+  if (loading) {
+    return <Spinner className="flex justify-center p-16" />;
+  }
+
+  if (error) {
+    return <ErrorBlock error={error} />;
   }
 
   if (
-    status === 'CONFIRMED' ||
-    status === 'REGISTER_FORM' ||
-    status === 'REGISTER_OVR'
+    needToVerifyVoterRegStatuses(registrationStatus) ||
+    verifiedRegistrationStatus
   ) {
     return (
-      <div className="voter-reg -green flex items-center">
-        <img className="post-badge icon-check" src={checkmark} alt="hello" />
-        <div className="ml-3">Your voter reg status is confirmed! Woo!</div>
+      <div
+        className={classnames('voter-reg flex items-center', {
+          '-green':
+            verifiedCompletedVoterRegStatuses(registrationStatus) ||
+            registrationStatus === 'CONFIRMED',
+        })}
+      >
+        {verifiedCompletedVoterRegStatuses(registrationStatus) ||
+        registrationStatus === 'CONFIRMED' ? (
+          <img
+            className="pl-2 post-badge icon-check"
+            src={checkmark}
+            alt="hello"
+          />
+        ) : null}
+        <div className="m-3">
```

Comment:
Nit: line break for sibling elements to help with readability
```suggestion

        <div className="m-3">
```